### ISSUE TITLE
Cant colcon build the pakage in the first place

### ISSUE THREAD

**Issue by Artinmi** [Muhammad540/traversability_estimation#2]
hi Muhammad thanks for porting this pkg on ROS2 , when i clone the package in my ros2 environment and try to do colcon build , i get to this error in Line 239 and 240 , ( then i tried to comment and ignore it , it will fix colcon build but when it comes to run the "ros2 launch traversability_estimation traversability_estimation.launch" it gives error about that two lines that we commented) so we cant comment Lines 239 and 240. but if you dont comment them you cant do colcon build. what should i do?
this the the function that contains line 239 and 240:
![Screenshot from 2024-11-23 14-43-27](https://github.com/user-attachments/assets/22017e28-7095-420c-8da4-9eb622cb8907)
and below is the error i get in colcon build if i dont comment lines 239 and 240:
![Screenshot from 2024-11-23 14-44-21](https://github.com/user-attachments/assets/716794ec-c6f3-46b9-92c7-88cb8e90780f)

**Comment by aakapatel** [comment]
Hi @Artinmi, In my implementation I have commented those lines. However I can still run the package. Can you send me snapshot of the error you get upon launching the node?

**Comment by Artinmi** [comment]
Yes of course , so you are saying that  we dont need those 2 lines and we can comment them. The error im get upon to when i launch the node is this: 
![389193210-716794ec-c6f3-46b9-92c7-88cb8e90780f](https://github.com/user-attachments/assets/ae2c804a-62a2-4f13-a2a7-b0c2040c6dab)

**Comment by Muhammad540** [comment]
@Artinmi I will look into this if i get time today. However, from the looks of it the issues seems to arise from grid map messages package. Can you confirm that  grid map messages dependency is built properly ? Since the "get grid map messages info" service does have header, [check the get grid map info srv structure](https://github.com/ANYbotics/grid_map/blob/master/grid_map_msgs/srv/GetGridMapInfo.srv)

**Comment by Artinmi** [comment]
thanks .I did what you said and im now sure that gridmap is working properly.i am facing another problem. I have changed all the  "base_link" parameters in travelsibility package and elevation mapping package  to "camera_link" (bacause im just testing with my realsense camera and dont have a robot right now ) .So the problem is despite the fact  that i have changed both elevation mapping and travelsibility parameters to camera_link but  when i run the launch file of travelsibility i get to this error saying that he is expect a map in "camera_link" frame_id but recieved it in "" frame_id : ( i will provide you the error , elevation mapping visulization , elevation parameters , travelsibility parameters that i have changed to camera_link)
![Screenshot from 2024-12-02 18-55-53](https://github.com/user-attachments/assets/59067bd6-49c6-450d-aac6-619d630e630f)
![Screenshot from 2024-12-02 18-58-17](https://github.com/user-attachments/assets/6c8ffa19-9ef9-4f30-baea-08f0ab12b8a8)
![Screenshot from 2024-12-02 19-11-47](https://github.com/user-attachments/assets/295a48fa-3dbe-4b45-838a-4a5004336e35)

![Screenshot from 2024-12-02 19-11-14](https://github.com/user-attachments/assets/53d572a0-0844-4ca3-8c63-bcc2d3a960b0)
![Screenshot from 2024-12-02 19-11-07](https://github.com/user-attachments/assets/f8d60cfd-a36d-4e53-8376-5849e2f7d02c)

**Comment by Muhammad540** [comment]
@Artinmi , nice to see that grid map issue is resolved. Firstly, I would suggest instead of setting all frames camera_link, try constructing a static transform tree from map to baselink to camera_link. This might not solve your immediate problem, but generally a good practice. So, focusing on your problem this is a brief overview of how this works: traversability estimation node is requesting the raw elevation map from the elevation mapping node, where the service is defined. I would suggest go to the elevation mapping service, as far as i remember it is called "get raw submap" and make sure that the message it is sending to client has the frame id correctly defined to be camera_link.
